### PR TITLE
[update] azure-llama-function-call-parsing

### PR DESCRIPTION
**Pull request by diogoazevedo15**
## Summary:
1. Refactor llama Function Call Parsing: The parsing logic for Llama function calls in `azure.py` has been updated to handle scenarios where functions are present but not used to generate the response.
2. Code Cleanup in provider.py: Removed redundant chunk-related code from `provider.py`.

### CODE REVIEW

File: llmstudio/engine/providers/azure.py [hunk 1]
```diff
@@ -174,7 +175,9 @@ async def handle_tool_response(
 
         function_call_buffer = ""
         saving = False
+        normal_call_chunks = []
         for chunk in response:
+            chunk
```
Comment (diogoncalves) [inline llmstudio/engine/providers/azure.py:180]: is this working? this line is not doing anything and this is not an inline condition.

File: llmstudio/engine/providers/azure.py [hunk 2]
```diff
@@ -433,14 +439,15 @@ def add_tool_instructions(self, tools: list) -> str:
         tool_prompt += """
```
Comment (diogoncalves) [inline llmstudio/engine/providers/azure.py:439]: should we move the prompts to a "default prompts" library?

File: llmstudio/engine/providers/provider.py [hunk 3]
```diff
@@ -268,28 +268,6 @@ def join_chunks(self, chunks, request):
                 ):
                     function_call_arguments += chunk.get("arguments")
 
-            chunk = ChatCompletion(
```
Comment (diogoncalves) [inline llmstudio/engine/providers/provider.py:271]: will this break other providers like openai, azureopen ai, etc?

**Review by diogoncalves — CHANGES_REQUESTED** [review]

File: llmstudio/engine/providers/provider.py (hunk 3 above)
Comment (diogoazevedo15) [reply llmstudio/engine/providers/provider.py:271]: This was included for debugging purposes and does not impact the code's behavior.

File: llmstudio/engine/providers/azure.py (hunk 2 above)
Comment (diogoazevedo15) [reply llmstudio/engine/providers/azure.py:439]: Yes, I like that suggestion. It's a bit outside the scope of this PR, so I'll open a new one to address it.

**Review by diogoncalves — APPROVED** [review]

**Pull request merged**